tests/test_run.py

```diff
@@ -44,22 +46,38 @@ def test_run_script_from_internet(pipx_temp_env, capsys):
     )
 
 
-def test_appargs_doubledash(capsys):
-    run_pipx_cli(["run", "pycowsay", "--", "hello"])
-    captured = capsys.readouterr()
-    assert "< -- hello >" not in captured.out
-    run_pipx_cli(["run", "pycowsay", "hello", "--"])
-    captured = capsys.readouterr()
-    assert "< hello -- >" not in captured.out
-    run_pipx_cli(["run", "pycowsay", "--"])
-    captured = capsys.readouterr()
-    assert "< -- >" not in captured.out
-    run_pipx_cli(["run", "--", "pycowsay", "--", "hello"])
-    captured = capsys.readouterr()
-    assert "< -- hello >" not in captured.out
-    run_pipx_cli(["run", "--", "pycowsay", "hello", "--"])
-    captured = capsys.readouterr()
-    assert "< hello -- >" not in captured.out
-    run_pipx_cli(["run", "--", "pycowsay", "--"])
-    captured = capsys.readouterr()
-    assert "< -- >" not in captured.out
+def test_appargs_doubledash(pipx_temp_env, capsys, monkeypatch):
+    parser = pipx.main.get_command_parser()
+
+    input_argv = [
+        ["pipx", "run", "pycowsay", "--", "hello"],
+        ["pipx", "run", "pycowsay", "--", "--", "hello"],
+        ["pipx", "run", "pycowsay", "hello", "--"],
+        ["pipx", "run", "pycowsay", "hello", "--", "--"],
+        ["pipx", "run", "pycowsay", "--"],
+        ["pipx", "run", "pycowsay", "--", "--"],
+        ["pipx", "run", "--", "pycowsay", "--", "hello"],
+        ["pipx", "run", "--", "pycowsay", "--", "--", "hello"],
+        ["pipx", "run", "--", "pycowsay", "hello", "--"],
+        ["pipx", "run", "--", "pycowsay", "hello", "--", "--"],
+        ["pipx", "run", "--", "pycowsay", "--"],
+        ["pipx", "run", "--", "pycowsay", "--", "--"],
+    ]
+    expected_appargs = [
+        ["--", "hello"],
+        ["--", "--", "hello"],
+        ["hello", "--"],
+        ["hello", "--", "--"],
+        ["--"],
+        ["--", "--"],
+        ["--", "hello"],
+        ["--", "--", "hello"],
+        ["hello", "--"],
+        ["hello", "--", "--"],
+        ["--"],
+        ["--", "--"],
+    ]
+    for i, input_argv in enumerate(input_argv):
+        monkeypatch.setattr(sys, "argv", input_argv)
+        parsed_pipx_args = pipx.main.parse_args(parser)
+        assert parsed_pipx_args.appargs == expected_appargs[i]
```

Comment:
This is awesome. pytest has a `parameterize` decorator that adds some nice reporting if something fails. Recommend using it, but not a big deal. 

https://docs.pytest.org/en/latest/parametrize.html

Reply:
Agreed, perfect case for a parametrized test. It'll allow each one to pass or fail individually.

Alternatively, zip would be my preferred way to keep input and output iterating in sync with each other.